### ISSUE TITLE
Add Support for Analog Input Directionality.

### ISSUE THREAD

**Issue by DataPlusProgram** [rsubtil/controller_icons#119]
In my project, I’ve implemented directionality for analog inputs as follows:

![image](https://github.com/user-attachments/assets/e365c1a9-2d46-4528-a782-265181c053d7)

I was wondering if it would be possible to add this feature to the main project?

The only caveat is that some users might prefer the standard icon instead of having it broken into the four cardinal directions. In such cases, a way to choose the desired behavior would need to be implemented.

**Comment by rsubtil** [comment]
(sorry, I "replied" to this issue last week, but seems like I forgot to click send :man_facepalming:)

I've been thinking of implementing a "flair" system that allows to decorate icons with contextual actions like these for any icon, heavily inspired by Kenney's icon pack.
![image](https://github.com/user-attachments/assets/f36be9e7-ace9-4c4b-89cf-ba159f0f2fae)

**Comment by DataPlusProgram** [comment]
Yes it would be necessary to do that to avoid having to create a ton of new assets.

I would purpose a system as follows based on the Kenny assets:
## Mode 1: Fully Restricted
This is the current way it works.
![r_stick](https://github.com/user-attachments/assets/c43c43cd-fc8a-48fa-92e1-3d11b6b50484)

## Mode 2: Axis Restricted
I don't know a use case for this yet so this mode could possibly be ommited.

| Name         | x-axis                                                                                 | y-axis                                                                                 |
|--------------|-----------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------|
| **Flair**    | ![flair_arrows_horizontal](https://github.com/user-attachments/assets/165b1462-11a4-4dd0-af8c-d55c35646cc7) | ![flair_arrows_vertical](https://github.com/user-attachments/assets/db1f831d-7650-4650-b8f5-b7636644c064) |
| **Example**  | ![example_x_axis](https://github.com/user-attachments/assets/d18400f3-1478-4ceb-b591-b0290a251e9d)          | ![example_y_axis](https://github.com/user-attachments/assets/db885b18-8da0-4262-85e2-33517d1c7d82)         |

## Mode 3: Unrestricted
This is what I use in my current project.
| Name         | Up                                                                                  | Down                                                                                 | Left                                                                                  | Right                                                                                 |
|--------------|-------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------|
| **Flair**    | ![flair_arrows_up](https://github.com/user-attachments/assets/d800b022-ab2c-40cc-8adb-15e4dd4ecea6)  | ![flair_arrows_down](https://github.com/user-attachments/assets/711559b4-bb35-43e9-be3f-5f0fd8a008ae) | ![flair_arrows_left](https://github.com/user-attachments/assets/e3fd24f0-5bbc-4bc6-b6f6-5f2664c6b55b) | ![flair_arrows_right](https://github.com/user-attachments/assets/63520d11-5d17-4d88-8f0f-c01850f40474) |
| **Example**  | ![example_up](https://github.com/user-attachments/assets/da8e239e-648a-4599-9b80-001699a2a323)      | ![example_down](https://github.com/user-attachments/assets/dca0b53b-3f8b-42b5-bae7-aa6e190187c6)     | ![example_left](https://github.com/user-attachments/assets/41ddcc3c-9792-4350-bb43-b9577c1d5424)     | ![example_right](https://github.com/user-attachments/assets/5d4faea8-13e2-43b7-9257-5601b194d9a8)     |
  
A bonus with the flairs is that numbers could be used to denote device ID optionally as well.
